src/qtcore/qml/elements/QtQuick/MouseArea.js

```diff
@@ -81,6 +83,16 @@ registerQmlType({
         self.containsMouse = false;
         self.exited();
     }
+    this.dom.onmouseover = function(e) {
+        self.containsMouse = true;
+        self.dom.style.cursor = cursorShapeToCSS();
+        self.entered();
+    }
+    this.dom.onmouseout = function(e) {
+        self.containsMouse = false;
+        self.dom.style.cursor = "auto";
+        self.exited();
+    }
```

Comment:
Indeed, changing style here doesn't make sense :(. I can't remember why I did such a change :). I think we can remove these lines (self.dom.style.cursor). Do you want me to remove these lines now ?